### ISSUE TITLE
Rule proposal: `prefer-date-now`

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#909]
Prefer `Date.now()` over `new Date().getTime()`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now

Because `Date.now()` is shorter and nicer.

## Fail

```js
new Date().getTime();
```

## Pass

```js
Date.now();
```

**Comment by fisker** [comment]
Fail

```js
+new Date;
```

**Comment by sindresorhus** [comment]
Well, if you want obscure cases:

- `+new Date()`
- `Number(new Date())`
- `new Date().valueOf()`
- `(new Date).valueOf()`

**Comment by papb** [comment]
Also `someVariable - new Date()`? Or any arithmetic operation on it?

**Comment by sindresorhus** [comment]
@papb 👍🏻

**Comment by fisker** [comment]
```js
BigInt(new Date())
```


> Or any arithmetic operation on it?

`+` can't

**Comment by sindresorhus** [comment]
This is now accepted.
